### PR TITLE
adding back manual init if you dont like the values

### PR DESCRIPTION
**Pull request by ohayon**

### CODE REVIEW

File: lib/fastlane/setup/setup_ios.rb [hunk 1]
```diff
@@ -52,6 +45,47 @@ def run
...
+      generate_appfile(manually: true)
+      detect_installed_tools # after copying the existing files
+      ask_to_enable_other_tools
+      FileUtils.mkdir(path)
+      generate_fastfile(manually: true)
+      show_analytics
+    end
+
+    def ask_to_enable_other_tools
+      create_app = agree('Would you like to create your app on iTunes Connect and the Developer Portal?', true)
```
Comment (KrauseFx) [inline lib/fastlane/setup/setup_ios.rb]: This question doesn't make sense to the user if the app is already there. We should probably rephrase it

File: lib/fastlane/setup/setup_ios.rb [hunk 2]
```diff
@@ -52,6 +45,47 @@ def run
...
+      if create_app
+        create_app_if_necessary
+      end
+      if create_app and agree("Do you want to setup 'deliver', which is used to upload app screenshots, app metadata and app updates to the App Store? This requires the app to be in the App Store already. (y/n)".yellow, true)
```
Comment (KrauseFx) [inline lib/fastlane/setup/setup_ios.rb]: `This requires the app to be in the App Store already.` isn't really true if the user said `y` on the previous question. how about we only ask the user for deliver if the user also created the app?

File: lib/fastlane/setup/setup_ios.rb [hunk 3]
```diff
@@ -52,6 +45,47 @@ def run
...
+        enable_deliver
+      end
+
+      if !@tools[:snapshot] and Helper.mac? and agree("Do you want to setup 'snapshot', which will help you to automatically take screenshots of your iOS app in all languages/devices? (y/n)".yellow, true)
```
Comment (KrauseFx) [inline lib/fastlane/setup/setup_ios.rb]: We should remove the `snapshot` code, it's also not there with the automatic mode, right?

File: lib/fastlane/setup/setup_ios.rb (hunk 2 above)
Comment (ohwutup) [inline lib/fastlane/setup/setup_ios.rb]: what is a good way to know for sure that they created the app?

File: lib/fastlane/setup/setup_ios.rb (hunk 1 above)
Comment (ohwutup) [inline lib/fastlane/setup/setup_ios.rb]: you're totally right, we should check if its there before asking right? what is the best way to check if the app exists on iTC? i was playing around with using produce and deliver to try to get confirmation that an app exists but didn't know if i was finding exactly what i was looking for.

Comment (KrauseFx) [inline lib/fastlane/setup/setup_ios.rb]: We already do that in `detect_if_app_is_available` :+1:

File: lib/fastlane/setup/setup_ios.rb (hunk 2 above)
Comment (KrauseFx) [inline lib/fastlane/setup/setup_ios.rb]: You just asked the user before, you can just move this code inside the `if` block

File: lib/fastlane/setup/setup_ios.rb [hunk 4]
```diff
@@ -52,6 +45,41 @@ def run
...
+      generate_appfile(manually: true)
+      detect_installed_tools # after copying the existing files
+      ask_to_enable_other_tools
+      FileUtils.mkdir(path)
+      generate_fastfile(manually: true)
+      show_analytics
+    end
+
+    def ask_to_enable_other_tools
+      unless self.itc_ref && self.portal_ref
```
Comment (KrauseFx) [inline lib/fastlane/setup/setup_ios.rb]: That's super confusing and might even be wrong. How is the priority of `unless` and `&&`

Comment (KrauseFx) [inline lib/fastlane/setup/setup_ios.rb]: Instead do something like
```ruby
if self.itc_ref.nil? || self.port_ref.nil?
```

File: lib/fastlane/setup/setup_ios.rb [hunk 5]
```diff
@@ -52,6 +45,41 @@ def run
...
+        wants_to_create_app = agree('Would you like to create your app on iTunes Connect and the Developer Portal?', true)
+        if wants_to_create_app
+          create_app_if_necessary
+          detect_if_app_is_available # check if the app was, in fact, created.
+        end
+      end
+      if self.itc_ref && self.portal_ref
```
Comment (KrauseFx) [inline lib/fastlane/setup/setup_ios.rb]: `create_app_if_necessary` doesn't set a ref, does it? So this won't actually work

Comment (ohwutup) [inline lib/fastlane/setup/setup_ios.rb]: thats why i added another call to `detect_if_app_is_available` after `create_app_if_necessary`

Comment (ohwutup) [inline lib/fastlane/setup/setup_ios.rb]: does that make sense @KrauseFx?

Comment (KrauseFx) [inline lib/fastlane/setup/setup_ios.rb]: oh perfect, yeah

**Comment by ohwutup** [comment]
@KrauseFx gave the :+1: over VC

**Pull request merged**